code/modules/clothing/neck/_neck.dm

```diff
@@ -177,16 +177,55 @@
 
 /obj/item/clothing/neck/petcollar
 	name = "pet collar"
-	desc = "It's for pets. Though you probably could wear it yourself, you'd doubtless be the subject of ridicule."
+	desc = "It's for pets. Though you probably could wear it yourself, you'd doubtless be the subject of ridicule. It seems to be made out of a polychromic material."
 	icon_state = "petcollar"
+	alternate_worn_icon = 'icons/mob/neck.dmi'
 	item_color = "petcollar"
+	hasprimary = TRUE
+	primary_color = "#00BBBB"
 	pocket_storage_component_path = /datum/component/storage/concrete/pockets/small/collar
 	var/tagname = null
 
 /obj/item/clothing/neck/petcollar/attack_self(mob/user)
 	tagname = copytext(sanitize(input(user, "Would you like to change the name on the tag?", "Name your new pet", "Spot") as null|text),1,MAX_NAME_LEN)
 	name = "[initial(name)] - [tagname]"
 
+/obj/item/clothing/neck/petcollar/worn_overlays(isinhands, icon_file)
+	. = ..()
+	if(hasprimary | hassecondary | hastertiary)
+		if(!isinhands)	//prevents the worn sprites from showing up if you're just holding them
+			if(hasprimary)	//checks if overlays are enabled
+				var/mutable_appearance/primary_worn = mutable_appearance(alternate_worn_icon, "[item_color]-primary")	//automagical sprite selection
+				primary_worn.color = primary_color	//colors the overlay
+				. += primary_worn	//adds the overlay onto the buffer list to draw on the mob sprite.
+			if(hassecondary)
+				var/mutable_appearance/secondary_worn = mutable_appearance(alternate_worn_icon, "[item_color]-secondary")
+				secondary_worn.color = secondary_color
+				. += secondary_worn
+			if(hastertiary)
+				var/mutable_appearance/tertiary_worn = mutable_appearance(alternate_worn_icon, "[item_color]-tertiary")
+				tertiary_worn.color = tertiary_color
+				. += tertiary_worn
```

Comment:
This entire proc as a whole could probably be made far more efficient by using color matrix compatible icons instead of relying on multiple layers of overlays. (this comment does not affect review, just a comment and not a requested change)

Reply:
Faaair. But I'm lazy as fuck. Someone sometime will hopefully update all that stuff.